examples/.experimental/next-app-dir/src/trpc/server.ts

```diff
@@ -20,21 +23,25 @@ export const api = experimental_createTRPCNextAppDirServer<AppRouter>({
         httpBatchLink({
           url: getUrl(),
           headers() {
-            const newHeaders = new Map(headers());
-
-            // If you're using Node 18 before 18.15.0, omit the "connection" header
-            newHeaders.delete('connection');
-
-            // `x-trpc-source` is not required, but can be useful for debugging
-            newHeaders.set('x-trpc-source', 'rsc');
-
-            // Forward headers from the browser to the API
-            return Object.fromEntries(newHeaders);
+            return {
+              cookie: cookies().toString(),
+              'x-trpc-source': 'rsc',
+            };
           },
         }),
       ],
     };
   },
 });
 
-// export const createAction =
+export const createAction = experimental_createServerActionHandler({
+  router: appRouter,
```

Comment:
signature 2 - pass in the router and then use the client proxy for actions too.

note: you can still use this signature and still inline action resolvers